src/app/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { useDispatch, useSelector } from 'react-redux';
+import type { AppDispatch, RootState } from './store';
+
+export const useAppSelector = useSelector.withTypes<RootState>();
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
```

Comment:
The `useSelector.withTypes` and `useDispatch.withTypes` methods do not exist in `react-redux`. Instead, you should use TypeScript's generic syntax to type these hooks. For example, `export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;` and `export const useAppDispatch = () => useDispatch<AppDispatch>();`.